core/executor/base_executor.go

```diff
@@ -40,6 +80,15 @@ func (e *BaseExecutor) VerifyInputs() error {
 			return err
 		}
 
+		gasAccountIndex, gasFeeAssetId, gasFeeAssetAmount, err := e.parseGas()
```

Comment:
I think we can add a method for all tx types to get its gasAccount, gasFeeAssetId, then no need to do this parse work again. Just a suggestion, you can decide which one is better.

Reply:
`gasAccountIndex`, `gasFeeAssetId`, `gasFeeAssetAmount` these three fields are standard fields. I think it is easy to parse them for now.